<launch>
   <!-- videre_mode should be one of the following:
         disparity:     Disparity and rectification done on chip.
                        Provides: Disparity and left mono image
         disparity_raw: Disparity done on chip (debayer and rectification in software).
                        Provides: disparity and left color image.
         none:          No stereo on chip (all processing done in software).
                        Provides: all 3 images available
         rectified:     Rectification on chip
                        Provides: all 3 images available but no color
          -->
  <!-- stereo processing parameters
       texture_thresh:  Threshold for removing bad disparities based on texture
                        Default value: 30
       unique_thresh:   Threshold for removing bad disparities based on ambiguity
                        Default value: 36
       speckle_diff:    Threshold for region-growing (1/16 pixel disparity)
                        Default value: 10
       speckle_size:    Threshold for disparity region size (pixels)
                        Default value: 100
       horopter:        X offset for close-in stereo (pixels)
                        Default value: 0
       corr_size:       Correlation window size (pixels)
                        Default value: 15
       num_disp:        Number of disparities (pixels)
                        Default value: 64
       -->
  <group ns="narrow_stereo">
    <node pkg="stereo_image_proc" type="stereoproc" respawn="false" output="screen" name="stereoproc">
      <param name="videre_mode" type="str" value="none"/>
      <param name="do_colorize" type="bool" value="True"/>
      <param name="do_rectify" type="bool" value="True"/>
      <param name="do_stereo" type="bool" value="True"/>
      <param name="do_calc_points" type="bool" value="True"/>
      <param name="do_keep_coords" type="bool" value="True"/>
      <param name="num_disp" type="int" value="128"/>
      <param name="corr_size" type="int" value="15"/>
    </node>
  </group>
</launch>

